corehq/apps/sms/tasks.py

```diff
@@ -129,6 +129,39 @@ def message_is_stale(msg, utcnow):
         return True
 
 
+def connection_slot_key_base(backend):
+    return 'backend-%s-connection-slot-' % backend.couch_id
+
+
+def reserve_connection_slot(backend, max_simultaneous_connections):
+    """
+    There is one redis key per connection slot, numbered from 1 to
+    max_simultaneous_connections.
+    A slot is considered taken if the corresponding redis key exists,
+    or is considered free if it does not exist.
+    """
+    with CriticalSection(['reserve-connection-slot-for-%s' % backend.couch_id]):
+        client = get_redis_client()
+        key_base = connection_slot_key_base(backend)
+        slot_keys = client.keys(key_base + '*')
```

Comment:
@gcapalbo I think this is causing issues with redis. Most of the entries in the redis slow log are from this:
```
128) 1) (integer) 52935
     2) (integer) 1516023372
     3) (integer) 957970  (microseconds = 0.95 s)
     4) 1) "KEYS"
        2) ":1:backend-7d74ce121f9d4d379d084278e4987cbf-connection-slot-*"
     5) "172.24.32.14:37935"
     6) ""
```

Also this is what the slowlog metric looks like which roughly coincides with when this was deployed:
![image](https://user-images.githubusercontent.com/249606/35039601-08a00b3a-fb87-11e7-956e-23b8bb59a51e.png)

Could you not use a redis set to do this?